Framework/API/src/MultiPeriodGroupWorker.cpp

```diff
@@ -203,12 +203,20 @@ bool MultiPeriodGroupWorker::processGroups(
   WorkspaceGroup_sptr outputWS = boost::make_shared<WorkspaceGroup>();
   AnalysisDataService::Instance().addOrReplace(outName, outputWS);
 
+  double progress_proportion = 1.0 / static_cast<double>(nPeriods);
   // Loop through all the periods. Create spawned algorithms of the same type as
   // this to process pairs from the input groups.
   for (size_t i = 0; i < nPeriods; ++i) {
     const int periodNumber = static_cast<int>(i + 1);
-    Algorithm_sptr alg_sptr = API::AlgorithmManager::Instance().createUnmanaged(
-        sourceAlg->name(), sourceAlg->version());
+    // use create Child Algorithm that look like this one
+    Algorithm_sptr alg_sptr = sourceAlg->createChildAlgorithm(
+        sourceAlg->name(), progress_proportion * periodNumber,
+        progress_proportion * (1 + periodNumber), sourceAlg->isLogging(),
+        sourceAlg->version());
+    // Don't make the new algorithm a child so that it's workspaces are stored
+    // correctly
+    alg_sptr->setChild(false);
+    alg_sptr->setRethrows(true);
```

Comment:
That is fair enough. 

Now the code seems to be going in different directions in different places:
- We have `Algorithm` , `MultiPeriodGroupWorker` and one algorithm `ReflectometryReductionOneAuto`) that	use the `setChild(false)` trick as here, and then we also have [this place](https://github.com/mantidproject/mantid/blob/master/Framework/PythonInterface/mantid/api/src/Exports/BinaryOperations.cpp#L164-L166) with a `createUnmanaged` followed by a `setChild(false)`.
- And then we have `DataProcessorAlgorithm` and some 6 algorithms that prefer to use `setAlwaysStoreInADS()`.